kcp/login.sh

```diff
@@ -0,0 +1,236 @@
+#!/bin/bash
+set -euo pipefail
+
+KUBECONFIG_KCP="${KUBECONFIG_KCP:-$HOME/.kube/kcp.config}"
+KUBECONFIG_KCP_ARGOCD="${KUBECONFIG_KCP_ARGOCD:-$HOME/.kube/kcp.argocd-manager.config}"
+KUBECONFIG_PLNSVC_CLUSTERADMIN="${KUBECONFIG_PLNSVC_CLUSTERADMIN:-$HOME/.kube/plnscv.clusteradmin.config}"
+
+
+usage(){
+  echo "
+Usage:
+    ${0##*/} [options]
+
+Log onto the clusters running on KCP.
+
+Optional arguments:
+    -k, --kcp KCP_CONTEXT
+        Select the KCP instance on which to login. Must be one of [stable, unstable].
+        Default value: unstable
+    -w, --workspace KCP_WORKSPACE
+        Select the KCP workspace to log into.
+        Default value: default
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    ${0##*/} --all
+" >&2
+}
+
+
+
+parse_args(){
+  KCP_ENV="kcp-unstable"
+
+  local args
+  args="$(getopt -o dhk:w: -l "debug,help,kcp,workspace" -n "$0" -- "$@")"
+  eval set -- "$args";
+  while true; do
+    case $1 in
+      -k|--kcp)
+        shift
+        case $1 in
+          stable|unstable)
+            KCP_ENV="kcp-$1"
+            ;;
+          *)
+            echo "Unsupported environment: $1" >&2
+            echo "" >&2
+            usage
+            exit 1
+            ;;
+        esac
+        ;;
+      -w|--workspace)
+        shift
+        KCP_WORKSPACE="$1"
+        ;;
+      -d|--debug)
+        set -x
+        ;;
+      -h|--help)
+        usage
+        exit 0
+        ;;
+      --)
+        # End of arguments
+        break
+      ;;
+      *)
+        echo "Unknown argument: $1"
+        usage
+        exit 1
+        ;;
+    esac
+    shift
+  done
+}
+
+
+
+generate_kcp_kubeconfig(){
```

Comment:
harcoding the entire kubeconfig this way may be risky going forward. We need to think how we could keep a track if kcp team changes this kubeconfig? (which has already happened in the recent past)
Maybe, we could request kcp team to put the kubeconfig into some code repo? just a thought.

Reply:
I've requested for the template to be put under source control (see [this comment](https://docs.google.com/document/d/1ygHQOBDMMnnsEDQoXSLSxaLUrplhOC42R3_kHCr25TI/edit?disco=AAAAWoIiN8o)).
The current solution should be good enough until the template is on git, and the fix willmbe quite easy.